### PR TITLE
Port to jbuilder

### PR DESCRIPTION
**Pull request by mseri**
This will need a SPEC file release with a new specfile

Signed-off-by: Marcello Seri <marcello.seri@citrix.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,43 +1,30 @@
...
 
-travis-coveralls.sh:
-	wget https://raw.githubusercontent.com/simonjbeaumont/ocaml-travis-coveralls/master/$@
+gh-pages:
+	bash .docgen.sh
 
-coverage: travis-coveralls.sh
-	bash $<
+reindent:
+	ocp-indent --syntax cstruct -i **/*.ml
```
Comment (kc284) [inline Makefile]: I think the recommendation now is to use git-ls to find the files:
`git ls-files '*.ml*' '**/*.ml*' | xargs ocp-indent --syntax cstruct -i`

Comment (mseri) [reply Makefile]: Indeed! These are all old, I keep forgetting to update the make file :D Thanks for spotting it

File: xapi-rrd-transport-utils.opam [hunk 2]
```diff
@@ -4,22 +4,10 @@ authors: "john.else@citrix.com"
...
-]
-build-test: [
-  [make "test"]
-]
-install: [
-  [make "PREFIX=%{prefix}%" "install"]
-]
-remove: [make "PREFIX=%{prefix}%" "uninstall"]
+build: [[ "jbuilder" "build" "-p" name "-j" jobs ]]
+build-test: ["jbuilder" "runtest" "-p" name]
```
Comment (kc284) [inline xapi-rrd-transport-utils.opam]: I think we need the ounit dependency since this opam file has a test directive.

Comment (mseri) [reply xapi-rrd-transport-utils.opam]: You are right

**Comment by mseri** [comment]
Now the build is working and coverage is back again.
Note that the build failure is currently due to some obscure failures in xcp-idl tests that are passing locally with the same set of packages

**Comment by mseri** [comment]
The current xapi-idl failure is not related to this change. The new ocaml-rpc makes a better json serialization that does not waste spaces. The tests convert something to jsonrpc and then to string, and compares the string with an expected value:
```
got ->      {"updates":[{"id":1,"v":"[\"Foo\",\"a\"]"}],"barriers":[[1,2,[{"id":1,"v":"[\"Foo\",\"a\"]"}]]]}
expected -> {"updates": [{"id": 1, "v": "[\"Foo\", \"a\"]"}], "barriers": [[1, 2, [{"id": 1, "v": "[\"Foo\", \"a\"]"}]]]}
```
Now it fails because jsonrpc does a better job

File: .coverage.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+set -ex
+
+COVERAGE_DIR=.coverage
+rm -rf $COVERAGE_DIR
+mkdir -p $COVERAGE_DIR
+pushd $COVERAGE_DIR
+if [ -z "$KEEP" ]; then trap "popd; rm -rf $COVERAGE_DIR" EXIT; fi
+
+$(which cp) -r ../* .
```
Comment (lindig) [inline .coverage.sh]: What is wrong with `/bin/cp` if there are any ambiguities? Could files also be moved instead of copied?

Comment (mseri) [reply .coverage.sh]: No idea, the script is quite old. I did only the minimal necessary changes for it to keep working. I think we need to find a better solution for coverage but I think it's out of scope for this PR

**Comment by mseri** [comment]
@lindig @kc284 if you are happy with this I'd merge it

**Pull request merged**